lib/tools.js

```diff
@@ -139,11 +137,45 @@ function asJson (obj, msg, num, time) {
           value = (stringifier || stringify)(value)
       }
       if (value === undefined) continue
-      data += ',"' + key + '":' + value
+      propStr += ',"' + key + '":' + value
+    }
+  }
+
+  let msgStr = ''
+  if (msg !== undefined) {
+    value = serializers[messageKey] ? serializers[messageKey](msg) : msg
+    const stringifier = stringifiers[messageKey] || wildcardStringifier
+
+    switch (typeof value) {
+      case 'function':
+        break
+      case 'number':
+        /* eslint no-fallthrough: "off" */
+        if (Number.isFinite(value) === false) {
+          value = null
+        }
+      // this case explicitly falls through to the next one
+      case 'boolean':
+        if (stringifier) value = stringifier(value)
+        msgStr = ',"' + messageKey + '":' + value
+        break
+      case 'string':
+        value = (stringifier || asString)(value)
+        msgStr = ',"' + messageKey + '":' + value
+        break
+      default:
+        value = (stringifier || stringify)(value)
+        msgStr = ',"' + messageKey + '":' + value
```

Comment:
Was this code lifted/copied from somewhere else?

Reply:
It's the handling of the values from above https://github.com/pinojs/pino/blob/83d80f63888aa0b68479d460706644c58257131c/lib/tools.js#L118

I duplicated it to avoid creating a separate function (for performance reasons).

But maybe all that handling is not needed for only the 'msg' key.

Reply:
Regrding the benchmarks registered benchmarks on the cloud, they seem to vary by a statistically significant ammount even in the other loggers:
<img width="1438" alt="image" src="https://user-images.githubusercontent.com/11346489/89331981-53a7c400-d69b-11ea-93eb-f3f43f83bb99.png">

But it's very late and I am tired so I might be overlooking something 😄.

I will investigate and get back to you.

Also, do you guys have a better tool other then diff for comparing the performance (maybe something that registers the % gain and the standard deviation of the ran tests)?

And I see the clould machine is an Azure, I guess it's a VM without guaranteed resources so the performance might vary over time (didn't use Azure a lot so I might be mistaken).

Reply:
The benchmarks that are automatically run by the PR process are run in GitHub's Actions platform. At the bottom of each file you will find a summary of the each result that is the average of all the runs for each result type.

Reply:
I ran the `update-bench-doc` task with both versions on my mac. The results seem to be almost the same (some times one version or the other is faster but not by something significant).

Left side is the old version and the right is the new PR.
![image](https://user-images.githubusercontent.com/11346489/89760992-22455300-daf6-11ea-9416-c75ad9a5a436.png)

I think the problem on the clould is the fact that it runs on a VM w/o guaranteed resources. This means performance can vary widly according to the usage of other VMs sharing the same underlying resources.

Do you guys agree? Maybe someone can double check on a phisical machine just to be sure.

Reply:
The VM resources are supposed to be the same across all runs -- https://docs.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners

Anyway, the benchmark action is merely there to provide a quick inspection. True benchmark comparisons need to be run on a dedicated system if the PR benchmarks give an indication that something may be amiss.

Reply:
So I think that this means the benchmarks are OK. I will resolve this conversation and ask for a re-review.